### PR TITLE
Implementing DPH/DPHI vertex shader opcodes

### PR DESCRIPTION
**Pull request by mailwl**
to avoid ASSERT_DEBUG for game "Tales of the Abyss"

### CODE REVIEW

**Comment by mailwl** [comment]
woops, i broke CI :(

File: src/video_core/vertex_shader.cpp [hunk 1]
```diff
@@ -230,6 +230,24 @@ static void ProcessShaderCode(VertexShaderState& state) {
                 break;
             }
 
+            case OpCode::Id::DPH:
+            case OpCode::Id::DPHI:
+            {
+                float24 dot = float24::FromFloat32(0.f);
+                for (int i = 0; i < 3; ++i)
+                    dot = dot + src1[i] * src2[i];
```
Comment (lioncash) [inline src/video_core/vertex_shader.cpp]: `dot += src1[i] * src2[i];`

File: src/video_core/vertex_shader.cpp [hunk 2]
```diff
@@ -230,6 +230,24 @@ static void ProcessShaderCode(VertexShaderState& state) {
...
+
+                dot = dot + src2[3];
```
Comment (lioncash) [inline src/video_core/vertex_shader.cpp]: `dot += src2[3];`

**Comment by Subv** [comment]
LGTM, please squash your commits.

**Comment by purpasmart96** [comment]
Is the forth component forced to 1.0?

**Comment by Subv** [comment]
3dbrew says that it's (src1.x, src1.y, src1.z, src1.w) * (src2.x, src2.y, src2.z, 1)

**Comment by mailwl** [comment]
it's from official nintendo documentation, i made similar to dp4 implemetation:
http://pastebin.com/zbFsakS5

**Comment by yuriks** [comment]
@mailwl If you're using official documentation you have either: committed copyright infringement by illegally downloading the SDK or committed a breach of contract by breaking your NDA. In either case, your contribution is of dubious legality and would implicate the project in copyright issues if merged, so I'm going to close this issue.

**Comment by MoochMcGee** [comment]
@yuriks Why are you pretending we haven't used those docs already? Why are
you pretending that other emulators are clean-room reverse-engineered?
Because you'd be wrong on both counts.

**Comment by yuriks** [comment]
@MoochMcGee I only stated facts. If anyone wants to come forward and say that they're a user of the SDK, they're welcome to do that, but we'll have to revert everything they ever contributed, and it would also open them to legal action. With that in mind, as a project it's not a position we can endorse without jeopardizing its mission.

**Comment by mailwl** [comment]
Sorry, don't know that. Vertex assembly is not confidencial, for example:
https://en.wikipedia.org/wiki/ARB_assembly_language

**Comment by bunnei** [comment]
@MoochMcGee we in fact do not use nor do we condone the use of Nintendo's SDK. Please consider this a firm verbal warning: Any further discussion regarding this subject will get you banned.

@mailwl as discussed above, we will not condone nor allow any contributions based off of copyrighted Nintendo material that was illegally obtained. That being said, I would encourage you to instead verify the behavior of these instructions on your own by writing your own homebrew test(s). If you can demonstrate that this implementation is correct in this manner, we will welcome your PR.

**Pull request closed without merging**